apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,260 @@
+        /// </summary>
+        /// <param name="year">The year (in the range 1-9999 inclusive).</param>
+        /// <param name="month">The month (in the range 1-12 inclusive).</param>
+        /// <param name="day">The day (1 through the number of days in <paramref name="month"/>).</param>
+        public SpannerDate(int year, int month, int day) : this(new DateTime(year, month, day))
+        {
+        }
+
+        private SpannerDate(int days) => _days = GaxPreconditions.CheckArgumentRange(days, nameof(days), 0, MaxNumberOfDays);
+
+        ///<inheritdoc/>
+        public int CompareTo(SpannerDate other) => _days.CompareTo(other._days);
+
+        /// <summary>
+        /// Implementation of <see cref="IComparable.CompareTo"/> to compare two <see cref="SpannerDate"/> values.
+        /// </summary>
+        /// <remarks>
+        /// This uses explicit interface implementation to avoid it being called accidentally. The generic implementation should usually be preferred.
+        /// </remarks>
+        /// <exception cref="ArgumentException"><paramref name="obj"/> is non-null but does not refer to an instance of <see cref="SpannerDate"/>.</exception>
+        /// <param name="obj">The object to compare this value with.</param>
+        /// <returns>The result of comparing this value with <paramref name="obj"/>. <paramref name="obj"/> is null, this method returns a value greater than 0.
+        /// </returns>
+        int IComparable.CompareTo(object obj)
+        {
+            if (obj == null)
+            {
+                return 1;
+            }
+            return obj is SpannerDate date
+                ? CompareTo(date)
+                : throw new ArgumentException($"Object must be of type {nameof(SpannerDate)}", nameof(obj));
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(SpannerDate other) => CompareTo(other) == 0;
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj) => obj is SpannerDate other && Equals(other);
+
+        /// <inheritdoc/>
+        public override int GetHashCode() => _days;
+
+        /// <summary>
+        /// Parses a textual representation in yyyy-MM-dd format as a <see cref="SpannerDate"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <paramref name="text"/> must be a representation of a date value in yyyy-MM-dd format which can be represented by <see cref="SpannerDate"/>.
+        /// </para>
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null. Must be in yyyy-MM-dd format only.</param>
+        /// <returns>The parsed value.</returns>
+        /// <exception cref="FormatException">The value could not be parsed as a <see cref="SpannerDate"/>.</exception>
+        public static SpannerDate Parse(string text)
+        {
+            string message = TryParseImpl(text, out var value);
+            return message == null ? value : throw new FormatException(message);
```

Comment:
Nit: prefer `message is null` over `message == null`. (We have a *lot* of `== null` through the codebase as it predates pattern matching, so don't worry too much :)